library/core/src/ptr/mod.rs

```diff
@@ -452,15 +452,10 @@ pub const unsafe fn swap<T>(x: *mut T, y: *mut T) {
 /// ```
 #[inline]
 #[stable(feature = "swap_nonoverlapping", since = "1.27.0")]
-pub unsafe fn swap_nonoverlapping<T>(x: *mut T, y: *mut T, count: usize) {
-    if cfg!(debug_assertions)
-        && !(is_aligned_and_not_null(x)
-            && is_aligned_and_not_null(y)
-            && is_nonoverlapping(x, y, count))
-    {
-        // Not panicking to keep codegen impact smaller.
-        abort();
-    }
+#[rustc_const_unstable(feature = "const_swap", issue = "none")]
+pub const unsafe fn swap_nonoverlapping<T>(x: *mut T, y: *mut T, count: usize) {
+    // FIXME: Is it okay to remove these checks? The same checks were removed for
+    // ptr::copy_nonoverlapping
```

Comment:
Do we want to keep these checks? The same checks were removed for `ptr::copy_nonoverlapping` when it was constified

Reply:
For `copy_nonoverlapping` we ended up removing them entirely also for performance reasons. I am not sure if those apply here.

Reply:
"Not sure" as in "Lets undo those changes for now so we can continue with the less controversial changes and maybe consider that for some other time"? Or just "Not sure" :)

Reply:
"Not sure" as in "let's see what others thing"... @oli-obk ?

Reply:
I'm for removing the checks. I don't see why `swap_nonoverlapping` should have them while `copy_nonoverlapping` doesn't have them.